python-client/giskard/scanner/performance/performance_bias_detector.py

```diff
@@ -153,23 +214,53 @@ def _detect_for_metric(
                 issue_info = PerformanceIssueInfo(
                     slice_fn=slice_fn,
                     metric=metric,
-                    metric_value_slice=metric_val,
-                    metric_value_reference=ref_metric_val,
+                    metric_value_slice=slice_metric.value,
+                    metric_value_reference=ref_metric.value,
                     slice_size=len(sliced_dataset),
                     threshold=self.threshold,
+                    p_value=p_value,
                 )
 
-                issues.append(
-                    PerformanceIssue(
-                        model,
-                        dataset,
-                        level=level,
-                        info=issue_info,
-                    )
-                )
+                issues.append(PerformanceIssue(model, dataset, level=level, info=issue_info))
 
-        return issues
+        return issues, p_values
 
 
 def _is_unbalanced_target(classes: pd.Series):
     return (classes.value_counts() / classes.count()).std() > 0.2
+
+
+def _calculate_slice_metrics(model, dataset, metric, slice_fn, with_pvalue=False):
+    sliced_dataset = dataset.slice(slice_fn)
+    slice_metric = metric(model, sliced_dataset)
+
+    if not with_pvalue:
+        return sliced_dataset, slice_metric, None
+
+    # Perform statistical tests
+    complementary_dataset = dataset.slice(lambda df: df[~df.index.isin(sliced_dataset.df.index)], row_level=False)
+    comp_metric = metric(model, complementary_dataset)
+    pvalue = None
```

Comment:
you don't necessarily have to initialize it here since it's down in every branch below